posthog/permissions.py

```diff
@@ -353,24 +353,31 @@ def has_permission(self, request, view) -> bool:
 
         return True
 
-    def check_team_and_org_permissions(self, request, view) -> None:
+    def check_team_and_org_permissions(
+        self,
+        request: Request,
+        view: View,
+    ) -> None:
+        if view.scope_object == "user":
+            return  # The /api/users/@me/ endpoint is exempt from team and org scoping
+
         scoped_organizations = request.successful_authenticator.personal_api_key.scoped_organizations
         scoped_teams = request.successful_authenticator.personal_api_key.scoped_teams
 
         if scoped_teams:
             try:
                 team = view.team
                 if team.id not in scoped_teams:
-                    raise PermissionDenied(f"API key does not have access to the requested project '{team.id}'")
-            except (ValueError, KeyError):
-                raise PermissionDenied(f"API key with scoped projects are only supported on project-based views.")
+                    raise PermissionDenied(f"API key does not have access to the requested project: ID {team.id}.")
+            except (KeyError, AttributeError):
+                raise PermissionDenied(f"API keys with scoped projects are only supported on project-based endpoints.")
```

Comment:
I think the reason I didn't do this is the response contains a bunch of info about the org and teams so you are basically granting read access to all teams... but sure 🤷

Reply:
There's definitely that read capability, but it's still governed by the `user` scope, and when you select that scope we do say "❗️ Note that the user object lists all organizations and projects you're in.", so I think this is a fair tradeoff